### ISSUE TITLE
Tripal 4 - Fix PHP 8.2 dynamic property deprecations

### ISSUE THREAD

**Issue by dsenalik** [tripal/tripal#1520]
# Core Tripal 4 Development Task

## Branch Name: tv4g0-issue1520-php8.2-deprecations

Pull #1519 adds PHP 8.2 to automated tests. This reveals some new deprecation notices, but most are related to [PHP 8.2: Dynamic Properties are deprecated](https://php.watch/versions/8.2/dynamic-properties-deprecated)

A random example from `tripal/src/Plugin/TripalIdSpace/TripalDefaultIdSpace.php` where the dynamically created property is `$this->messageLogger`
```
  /**
   * {@inheritdoc}
   */
  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
    parent::__construct($configuration, $plugin_id, $plugin_definition);

    // Instantiate the TripalLogger
    $this->messageLogger = \Drupal::service('tripal.logger');

  }
```

**Comment by laceysanderson** [comment]
Fixed by PR. Confirmed with @dsenalik that he is making separate issues for the other issues.

**Comment by dsenalik** [comment]
Reopening because I found another one, in `tripal/tripal.install` line 905 ` $messenger->addMessage('Removed PostgreSQL schema clonging functions');`

I did a recursive grep for other similar instances and did not find any.

**Comment by laceysanderson** [comment]
Good catch -merged and re-closed 🙃

**Comment by dsenalik** [comment]
There are more! This one is observed when creating a new organism `/bio_data/add/bio_data_1`

`Deprecated function: Creation of dynamic property Drupal\tripal_chado\Plugin\TripalIdSpace\ChadoIdSpace::$connection is deprecated in Drupal\tripal_chado\Plugin\TripalIdSpace\ChadoIdSpace->__construct() (line 83 of modules/contrib/tripal/tripal_chado/src/Plugin/TripalIdSpace/ChadoIdSpace.php). `

Fix this one with Pull #1543

**Comment by dsenalik** [comment]
Reopening because of another deprecation `Creation of dynamic property Drupal\tripal_chado\ChadoCustomTables\ChadoMview::$custom_table is deprecated`, see https://github.com/tripal/tripal/pull/1610#issuecomment-1688842412

Deal with this in pull #1617

**Comment by dsenalik** [comment]
Running all automated tests locally on PHP8.2 to find any other instances:

`Test 'Drupal\Tests\tripal_chado\Functional\TaxonomyImporterTest::testTaxonomyImporterSimpleTest' started`

`Deprecated: a:5:{s:11:"deprecation";s:78:"strcmp(): Passing null to parameter #1 ($string1) of type string is deprecated";s:5:"class";s:57:"Drupal\Tests\tripal_chado\Functional\TaxonomyImporterTest";s:6:"method";s:30:"testTaxonomyImporterSimpleTest";s:15:"triggering_file";s:91:"/var/www/drupal9/web/modules/contrib/tripal/tripal_chado/src/api/tripal_chado.query.api.php";s:11:"files_stack";a:11:{i:0;s:91:"/var/www/drupal9/web/modules/contrib/tripal/tripal_chado/src/api/tripal_chado.query.api.php";i:1;s:95:"/var/www/drupal9/web/modules/contrib/tripal/tripal_chado/src/api/tripal_chado.phylotree.api.php";i:2;s:103:"/var/www/drupal9/web/modules/contrib/tripal/tripal_chado/src/Plugin/TripalImporter/TaxonomyImporter.php";i:3;s:103:"/var/www/drupal9/web/modules/contrib/tripal/tripal_chado/src/Plugin/TripalImporter/TaxonomyImporter.php";i:4;s:109:"/var/www/drupal9/web/modules/contrib/tripal/tripal_chado/tests/src/Functional/Plugin/TaxonomyImporterTest.php";i:5;s:50:"/var/www/drupal9/web/sites/simpletest/TestCase.php";i:6;s:50:"/var/www/drupal9/web/sites/simpletest/TestCase.php";i:7;s:68:"/var/www/drupal9/vendor/phpunit/phpunit/src/Framework/TestResult.php";i:8;s:50:"/var/www/drupal9/web/sites/simpletest/TestCase.php";i:9;s:19:"Standard input code";i:10;s:19:"Standard input code";}} in /var/www/drupal9/vendor/symfony/phpunit-bridge/Legacy/SymfonyTestsListenerTrait.php on line 284`

`Test 'Drupal\Tests\tripal_chado\Functional\TaxonomyImporterTest::testTaxonomyImporterSimpleTest' ended`

So the code to insert a phylotree has an optional `analysis_id`. If it is not specified, then it is set to `NULL`. Then this happens in `tripal_chado/src/api/tripal_chado.phylotree.api.php`
```
  $values = [
    'analysis_id' => $options['analysis_id'],    <-- this is NULL when not specified
    'name' => $options['name'],
    'dbxref_id' => $options['dbxref_id'],
    'comment' => $options['description'],
    'type_id' => $options['type_id'],
  ];

  $phylotree = chado_insert_record('phylotree', $values, [], $schema_name);
```
but `chado_insert_record()` doesn't handle the NULL nicely. I think it would be appropriate to fix it there, so this is also fixed in pull #1617

**Comment by dsenalik** [comment]
The remaining problem running automated tests is likely the same problem as described in issue #1523
It starts with
```
There were 11 failures:

1) Drupal\Tests\tripal_chado\Functional\chadoInstallerTest::testInstallSchema
The schema we just installed was not in the database.
Failed asserting that false matches expected 1.
```
Here we are looking for `testchado` but the only schema actually installed is named `chado`. That probably explains the remaining ten failures.

This will be pursued in issue #1523
